### PR TITLE
Refactor linters and add name 

### PR DESCRIPTION
**Pull request by stevenjack**
### Problem

Needed more flexibility around running a set of linters for a given file as we may want to run multiple linters such as if resources are correct __and__ name is in correct format.

### Solution

* Create `Linter` type for functions that can act like a linter.
* Refactor existing linters out of `lint_directory.go` into `linters.go`.
* Add `LintNames` as default linter then add more linters for each file type, then look over each and call function with config.

### Notes

- **Remember** to tag your merge commit after merging 👍
- **Remember** to add the Clubhouse ticket url as a comment ♣️ 🏠

### CODE REVIEW

**Comment by stevenjack** [comment]
https://app.clubhouse.io/vidsy/story/6762

File: terraform/linter/linters.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+				variable.Name,
+			)
+		}
+	}
+
+	return nil
+}
+
+func isValidName(name string) bool {
+	return !strings.Contains(name, "-")
```
Comment (revett) [inline terraform/linter/linters.go]: 🏄 

Use a regex `a-z_`

File: terraform/linter/linters.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/config"
+	"github.com/pkg/errors"
+)
+
+const (
+	validNameRegex = `^[a-z0-9_]+$`
```
Comment (revett) [inline terraform/linter/linters.go]: Can we extend it to not allow `_` as a prefix or suffix, and only allow a single `_` at a time.

Example: https://regex101.com/r/BeBFQx/1

**Review by revett — CHANGES_REQUESTED** [review]
Comment on regex

**Review by revett — APPROVED** [review]
🍰

**Pull request merged**